File: detail/1. basic/11) 클래스 vs 객체 vs 인스턴스/Person.java
public class Person {
    // 멤버변수(속성)
    String name;
    int IQ;
    int str;

    // constructor
    public Person(String name, int IQ, int str) {
        this.name = name;
        this.IQ = IQ;
        this.str = str;
    }

    // Person a = new Person(); 처럼 아무런 매개변수 선언 안할 때 디폴트로 생성되는 객체
    public Person() {
        this.name = "alanwalker";
        this.IQ = 100;
        this.str = 100;
    }

    // 메서드
    public void levelup() {
        this.IQ = this.IQ + 1;
        this.str = this.str + 1;
        System.out.println(this.name + "의 지능과 힘이 증가했습니다!" + this.IQ + " / " + this.str);
    }

    public static void main(String[] args) {
        Person a = new Person(); // 객체 >> 인스턴스
        a.levelup();
        Person b; // 객체
        b = new Person("홍철", 1, 1000); // 인스턴스
        b.levelup();
    }
}

File: detail/1. basic/11) 클래스 vs 객체 vs 인스턴스/Person.js
class Person {
  constructor(name = "alanwalker", IQ = 100, str = 100) {
    this.name = name;
    this.IQ = IQ;
    this.str = str;
  }
  levelup() {
    this.IQ++;
    this.str++;
    console.log(
      this.name + "의 지능과 힘이 증가했습니다! " + this.IQ + " / " + this.str
    );
  }
}
const a = new Person(); // 디폴트인 alanwalker 인스턴스가 생성된다
a.levelup();
const b = new Person("홍철", 100, 1000);
b.levelup();
